### DISCUSSION TITLE
Build Script Question

### DISCUSSION THREAD

**Discussion by Bloofers** [vb-consulting/RazorSvelte#6] [Q&A] [answered]
Hello VB Consulting,

First off, I am hugely grateful for this template. I've been stuck in corporate .Net Framework for awhile but have been looking to add Svelte to one of my hybrid spa projects and didn't want a NodeJS backend. This is the closest thing I've found so far and it looks awesome!
I do have a question about building and deploying the project. I'm used to simple builds and deploys from Visual Studio, so forgive my ignorance, but will I have to use NodeJS to compile the Rollup/Svelte code each time I want to see markup/js changes?

**Comment by vbilopav** [comment, marked as answer]
Yes, the entire build system runs on NodeJS, so you will have to have NodeJS and NPM to run build for your entire frontend build. (JavaScript and CSS outputs).

Rollup bundler and minified plugins, Svelte compiler and SCSS compiler - they all use NodeJS.

I wish they could rewrite it in something else like Rust or Go.

However, that's only for building, you don't need NodeJS to run. There's is no NodeJS runtime, everything runs in your browser, backed by asp.ner backend server.

**Reply by Bloofers** [reply]
Thanks for the quick reply and thanks again for creating this template!
